protobuf, dvc, mitmproxy: depend on `six` as a formula

carlocab opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

See https://github.com/Homebrew/homebrew-core/issues/73875.

This is a follow up to #74909.

carlocab commented:
Looks like `mitmproxy` needs an older version of `protobuf`. Dropping it.

Rylan12 reviewed (commented):
Should `mitmproxy` have a revision bump, too? Otherwise 👍

carlocab commented:
Good catch. I thought I had added the revision bump, but that was when I removed the `protobuf` resource in favour of the formula (which didn't work). Must've forgot to put it back when I amended those changes.

Will fix this in a bit.

SMillerDev commented on Formula/protobuf.rb:
```diff
@@ -26,11 +27,7 @@ class Protobuf < Formula
-    url "https://files.pythonhosted.org/packages/6b/34/415834bfdafca3c5f451532e8a8d9ba89a21c9743a0c59fbd0205c7f9426/six-1.15.0.tar.gz"
-    sha256 "30639c035cdb23534cd4aa2dd52c3bf48f06e5f4a941509c8bafd8ce11080259"
-  end
+  depends_on "six"
```
How can it use six at runtime without python?

carlocab replied:
Presumably it uses six when you use the Python bindings. Otherwise the existing resource wouldn't need to be installed into `libexec`.

Rylan12 replied:
Maybe I'm missing something, but why does this change whether `python@3.9` is a runtime dependency or not?

carlocab replied:
Adding a `depends_on "six"` will require installing `python@3.9` when you install `protobuf` (because `six` `depends_on "python@3.9"`). Without this change, you can install `protobuf` without `python@3.9` installed.

Rylan12 replied:
Oh, duh 🤦

carlocab commented:
Still not sure what I think about `python@3.9` now becoming a hard `protobuf` dependency.

carlocab commented:
Will come back to this when #77373 is merged.

carlocab commented:
We build `six` for multiple Python versions now, so it no longer depends on Python at runtime. (#77513) Fine with adding this dep to `protobuf` now.
